common/expression/src/function.rs

```diff
@@ -765,3 +815,62 @@ pub fn vectorize_with_writer_passthrough_nullable_2_arg<I1: ArgType, I2: ArgType
         }
     }
 }
+
+pub fn vectorize_string_2_string(
+    func: impl Fn(<StringType as ValueType>::ScalarRef<'_>, &mut [u8]) -> Result<usize, String> + Copy,
+    estimate_bytes_fn: impl Fn((&[u8], &Buffer<u64>)) -> usize + Copy,
+) -> impl Fn(ValueRef<StringType>, &GenericMap) -> Result<Value<StringType>, String> + Copy {
+    move |arg1, _| match arg1 {
+        ValueRef::Scalar(val) => {
+            let offsets = vec![0, val.len() as u64];
+            let data_len = estimate_bytes_fn((val, &offsets.into()));
+            let builder =
+                StringColumnBuilder::try_from_transform_scalar(val, data_len, |val, buf| {
+                    func(val, buf)
+                })?;
+            Ok(Value::Scalar(builder.build_scalar()))
+        }
+        ValueRef::Column(col) => {
+            let iter = StringType::iter_column(&col);
+            let (data, offsets) = &col;
+            let data_len = estimate_bytes_fn((data, offsets));
+            let builder =
+                StringColumnBuilder::try_from_transform_column(iter, data_len, |val, buf| {
+                    func(val, buf)
+                })?;
+            Ok(Value::Column(builder.build()))
+        }
+    }
+}
+
+pub fn vectorize_passthrough_nullable_string_2_string(
+    func: impl Fn(<StringType as ValueType>::ScalarRef<'_>, &mut [u8]) -> Result<usize, String> + Copy,
+    estimate_bytes_fn: impl Fn((&[u8], &Buffer<u64>)) -> usize + Copy,
+) -> impl Fn(
+    ValueRef<NullableType<StringType>>,
+    &GenericMap,
+) -> Result<Value<NullableType<StringType>>, String>
++ Copy {
+    move |arg1, _| match arg1 {
+        ValueRef::Scalar(None) => Ok(Value::Scalar(None)),
+        ValueRef::Scalar(Some(val)) => {
+            let offsets = vec![0, val.len() as u64];
+            let data_len = estimate_bytes_fn((val, &offsets.into()));
+            let builder =
```

Comment:
Scalar should be mapped to `Scalar` result.

Reply:
Yes, it will return Scalar result. The offsets is only used to get `estimate_bytes`.

I do not want to match `ValueRef` in `estimate_bytes_fn`, so all `scalars` will cast to `one item` column, then pass to the `estimate_bytes_fn`. This can make `estimate_bytes_fn` simple.

The `estimate_bytes_fn` is like:
```rust
Fn((&[u8], &Buffer<u64>)) -> usize + Copy,
```